### PR TITLE
Upgrade to werkzeug 2.0.2

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé
This PR updates werkzeug to 2.0.2 to get us closer to the latest version and helps us to update transitive dependencies.

No app code is changed in this PR, only test code.

As part of this work, the following test dependencies were also updated:
- isort
- pytest
- pytest-mock
- pytest-xdist
- beautifulsoup4
- freezegun
- flake8
- flake8-print
- requests-mock
- jinja2-cli[yaml]

### CODE REVIEW

File: requirements-app.txt [hunk 1]
```diff
@@ -43,6 +43,7 @@ awscli-cwlogs>=1.4.6,<1.5
 # Putting upgrade on hold due to v1.0.0 using sha512 instead of sha1 by default
 itsdangerous==0.24  # pyup: <1.0.0
 
-git+https://github.com/cds-snc/notifier-utils.git@46.5.0#egg=notifications-utils
+#git+https://github.com/cds-snc/notifier-utils.git@46.5.0#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@81e2fcdcb7c4d2ce3814af6db2ceb0ccd8f258f1
```
Comment (jimleroyer) [inline requirements-app.txt:47]: Same comment here as in API around tagging notification-utils following [this document's instructions](https://docs.google.com/document/d/1CMCR4wiTisA3qZdkomuk0zriqeyVp-od5ZuUO95wATM/edit).

File: requirements.txt [hunk 2]
```diff
@@ -70,23 +71,23 @@ idna==2.10
...
 pyexcel-ezodf==0.3.4
-PyJWT==2.2.0
-pyparsing==2.4.7
+PyJWT==2.4.0
+pyparsing==3.0.9
 PyPDF2==1.26.0
 python-dateutil==2.8.2
 python-json-logger==2.0.1
-redis==3.5.3
+redis==4.3.4
```
Comment (jimleroyer) [inline requirements.txt:90]: This is a major version bump. We are sure it works as before, api breaking changes, etc?

File: tests/app/main/views/test_sign_in.py [hunk 3]
```diff
@@ -137,7 +138,7 @@ def test_process_sms_auth_sign_in_return_2fa_template(
     mock_verify_password.assert_called_with(
         api_user_active["id"],
         password,
-        {"location": None, "user-agent": "werkzeug/1.0.1"},
+        {"location": None, "user-agent": mock.ANY},
```
Comment (jimleroyer) [inline tests/app/main/views/test_sign_in.py:141]: 👍

File: tests/app/test_template_previews.py [hunk 4]
```diff
@@ -60,6 +61,9 @@ def test_from_database_object_makes_request(
     expected_filename,
     mock_get_service_letter_template,
 ):
+    # This test is calling `current_service` outside a Flask endpoint, so we need to make sure
+    # `service` is in the `_request_ctx_stack` to avoid an error
+    load_service_before_request()
```
Comment (jimleroyer) [inline tests/app/test_template_previews.py:66]: Did GDS go the same way with this code?

**Review by jimleroyer — APPROVED** [review]
Looks great overall. Let's change the notification-utils version to a tag though before shipping it. 🙏

File: tests/app/test_template_previews.py (hunk 4 above)
Comment (andrewleith) [reply tests/app/test_template_previews.py:66]: Yes, this is what they did.

File: requirements.txt (hunk 2 above)
Comment (andrewleith) [reply requirements.txt:90]: Hmm I'm not sure what precipitating this particular update.  Let me dig into this.

Comment (andrewleith) [reply requirements.txt:90]: `redis` gets installed by `flask-redis` but I'm not clear why this version bumped now.  I'll pin it for now back at 3.5.3 until we're ready to upgrade it.

**Review by jimleroyer — APPROVED** [review]
:shipit:

**Pull request merged**